### PR TITLE
Return correct message for no callback url

### LINKED ISSUE #946: OAuth: Useless Error With No Callback URL
1) Create an app
2) Attach an integration via OAuth
3) Attempt to Authenticate
4) Error about invalid signed data. Instead, we should put an error on screen that says a callback URL is required. In fact, authentication shouldn't even begin without a valid callback URL.

### PR DESCRIPTION
**Pull request by zipy124**
## What does this PR do?
Gives correct error message for no callback url specified.

## ✅ Pull Request checklist

- [x] Is this code complete?
- [x] Are tests done/modified?
- [x] Are docs written for this PR? (if applicable)

## 🚨 Is this a breaking change for API clients?
No

## :squirrel: Deploy notes
none

## Anything else
closes #946

### CODE REVIEW

**Comment by zipy124** [comment]
Potentially should add tests to demonstrate this behaviour, in fact i'll do that now!

**Comment by zipy124** [comment]
Potentially we should add documentation so users know what to put as the callback url?

File: backend/uclapi/oauth/views.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def authorise(request):
         response.status_code = 400
         return response
 
-    if app.callback_url is None:
+    if app.callback_url is None or app.callback_url.strip() == "":
```
Comment (ChristopherHammond13) [inline backend/uclapi/oauth/views.py]: Nice catch there!

File: backend/uclapi/oauth/views.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def authorise(request):
...
         response = PrettyJsonResponse({
             "ok": False,
             "error": "No callback URL set for this app."
```
Comment (ChristopherHammond13) [inline backend/uclapi/oauth/views.py]: I'm thinking this should be something that's better for the end user too. I'll write something better up.

**Comment by ChristopherHammond13** [comment]
@zipy124 do you like my error message? If so I'll merge this. Thanks so much for your help! This bug has annoyed a LOT of people 🤣

**Comment by zipy124** [comment]
> @zipy124 do you like my error message? If so I'll merge this. Thanks so much for your help! This bug has annoyed a LOT of people rofl

@ChristopherHammond13 Yes this is much better, I'm one of them i remember it bugging me and harry for a while during porticode :laughing:

**Comment by ChristopherHammond13** [comment]
> 
> 
> > @zipy124 do you like my error message? If so I'll merge this. Thanks so much for your help! This bug has annoyed a LOT of people rofl
> 
> @ChristopherHammond13 Yes this is much better, I'm one of them i remember it bugging me and harry for a while during porticode 😆

ha! Well bother you any more it shall not!

**Pull request merged**